### PR TITLE
Add support for spot, vault, and class transfers to exchange client

### PR DESCRIPTION
**Pull request by sree-revoori1**

### CODE REVIEW

File: src/errors.rs [hunk 1]
```diff
@@ -61,4 +61,6 @@ pub enum Error {
     NoCloid,
     #[error("ECDSA signature failed: {0:?}")]
     SignatureFailure(String),
+    #[error("Vault address not provided")]
```
Comment (lmlmt) [inline src/errors.rs:64]: nit: align error message with name for consistency, either `VaultAddressNotProvided` and `"Vault address not provided"` or `VaultAddressNotFound` and `"Vault address not found"`

File: src/exchange/exchange_client.rs [hunk 2]
```diff
@@ -172,6 +175,56 @@ impl ExchangeClient {
         self.post(action, signature, timestamp).await
     }
 
+    pub async fn class_transfer(
+        &self,
+        usdc: f64,
+        to_perp: bool,
+        wallet: Option<&LocalWallet>,
+    ) -> Result<ExchangeResponseStatus> {
+        let usdc = (usdc * 1e6).round() as u64;
```
Comment (lmlmt) [inline src/exchange/exchange_client.rs:184]: nit: add comment for what this means, i.e. payload expects usdc without decimals

File: src/exchange/exchange_client.rs [hunk 3]
```diff
@@ -172,6 +175,56 @@ impl ExchangeClient {
...
+        let timestamp = next_nonce();
+
+        let action = Actions::VaultTransfer(VaultTransfer {
+            vault_address,
+            is_deposit,
+            usd,
+        });
+        let connection_id = action.hash(timestamp, self.vault_address)?;
+        let action = serde_json::to_value(&action).map_err(|e| Error::JsonParse(e.to_string()))?;
+        let is_mainnet = self.http_client.base_url == BaseUrl::Mainnet.get_url();
```
Comment (lmlmt) [inline src/exchange/exchange_client.rs:222]: nit: can you define `HttpClient::is_mainnet`?

File: src/exchange/exchange_client.rs [hunk 4]
```diff
@@ -395,6 +448,36 @@ impl ExchangeClient {
...
+
+    pub async fn spot_transfer(
+        &self,
+        amount: &str,
+        destination: &str,
+        token: &str,
+        wallet: Option<&LocalWallet>,
+    ) -> Result<ExchangeResponseStatus> {
+        let wallet = wallet.unwrap_or(&self.wallet);
+        let hyperliquid_chain = if self.http_client.base_url.eq(MAINNET_API_URL) {
```
Comment (lmlmt) [inline src/exchange/exchange_client.rs:460]: and use `HttpClient::is_mainnet` here?

File: src/exchange/exchange_client.rs (hunk 3 above)
Comment (sree-revoori1) [reply src/exchange/exchange_client.rs:222]: Done

File: src/exchange/exchange_client.rs (hunk 4 above)
Comment (sree-revoori1) [reply src/exchange/exchange_client.rs:460]: Done

File: src/errors.rs (hunk 1 above)
Comment (sree-revoori1) [reply src/errors.rs:64]: Done

File: src/exchange/exchange_client.rs (hunk 2 above)
Comment (sree-revoori1) [reply src/exchange/exchange_client.rs:184]: Done

**Review by lmlmt — APPROVED** [review]

**Pull request merged**